label/label.go

```diff
@@ -90,6 +89,53 @@ func (m *GormLabelRepository) Create(ctx context.Context, u *Label) error {
 	return nil
 }
 
+// Save update the given label
+func (m *GormLabelRepository) Save(ctx context.Context, l Label) (*Label, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "label", "save"}, time.Now())
+	lbl := Label{}
+	tx := m.db.Where("id = ?", l.ID).First(&lbl)
+	oldVersion := l.Version
+	l.Version = lbl.Version + 1
+	if tx.RecordNotFound() {
+		log.Error(ctx, map[string]interface{}{
+			"label_id": l.ID,
+		}, "label cannot be found")
+		return nil, errors.NewNotFoundError("label", l.ID.String())
+	}
+	if err := tx.Error; err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"label_id": l.ID,
+			"err":      err,
+		}, "unknown error happened when searching the label")
+		return nil, errors.NewInternalError(ctx, err)
+	}
+	tx = tx.Where("Version = ?", oldVersion).Save(&l)
+	if err := tx.Error; err != nil {
+		// combination of name and space ID should be unique
+		if gormsupport.IsUniqueViolation(err, "labels_name_space_id_unique_idx") {
+			log.Error(ctx, map[string]interface{}{
+				"err":      err,
+				"name":     l.Name,
+				"space_id": l.SpaceID,
+			}, "unable to create label because a label with same already exists in the space")
+			return nil, errors.NewDataConflictError(fmt.Sprintf("label already exists with name = %s , space_id = %s", l.Name, l.SpaceID.String()))
```

Comment:
When we run into a similar unique violation in the space we return a NewBadParameterError instead. Maybe check in other repos what we do and report.

Reply:
In some places we are using `NewDataConflictError`

Reply:
@baijum I've created an issue #1794 to discuss this.